Client/Frontend/Browser/FaviconManager.swift

```diff
@@ -31,14 +36,81 @@ class FaviconManager : TabHelper {
+        internal var description: String {
+            return "No Image Loaded"
+        }
+    }
+    
+    private func loadFavicons(tab: Tab, profile: Profile, favicons: [Favicon]) -> Deferred<Maybe<[Favicon]>> {
+        let deferred = Deferred<Maybe<[Favicon]>>()
+        var deferreds: [() -> Deferred<Maybe<Favicon>>]
+        deferreds = favicons.map { favicon in
+            return { () -> Deferred<Maybe<Favicon>> in
+                if let url = NSURL(string: favicon.url) {
+                    return self.getFavicon(tab, iconUrl: url, icon: favicon, profile: profile)
+                }
+                else {
+                    return deferMaybe(FaviconError())
+                }
+            }
+        }
+        
+        accumulate(deferreds) >>== { favicons in
+                deferred.fill(Maybe(success: favicons))
+        }
+        
+        return deferred
+    }
+    
+    func getFavicon(tab: Tab, iconUrl: NSURL, icon: Favicon, profile: Profile) -> Deferred<Maybe<Favicon>> {
+        let deferred = Deferred<Maybe<Favicon>>()
+        let manager = SDWebImageManager.sharedManager()
+        let options = tab.isPrivate ?
+            [SDWebImageOptions.LowPriority, SDWebImageOptions.CacheMemoryOnly] : [SDWebImageOptions.LowPriority]
+        
+        if let currentURL = tab.url,
+            let url = tab.url?.absoluteString {
+            let site = Site(url: url, title: "")
+            manager.downloadImageWithURL(iconUrl,
+                                         options: SDWebImageOptions(options),
+                                         progress: nil,
+                                         completed: { (img, err, cacheType, success, url) -> Void in
+                                            let fav = Favicon(url: url.absoluteString,
+                                                date: NSDate(),
+                                                type: icon.type)
+                                            
+                                            if let img = img {
+                                                fav.width = Int(img.size.width)
+                                                fav.height = Int(img.size.height)
+                                            } else {
+                                                deferred.fill(Maybe(failure: FaviconError()))
+                                                return
+                                            }
+                                            
+                                            tab.favicons.append(fav)
+                                            if !tab.isPrivate {
+                                                self.profile.favicons.addFavicon(fav, forSite: site).upon { _ in
+                                                    deferred.fill(Maybe(success: fav))
+                                                }
+                                                if tab.favicons.isEmpty {
```

Comment:
This check will always fail because we append the favicon to `tab.favicons` a few lines up. I would double check with @jhugman about when/how we're supposed to use the `makeFaviconAvailable` method since it relates to the Spotlight extension.